Improve full width components (masthead and primary navigation) (#192)
x-govuk/govuk-prototype-components: docs/masthead.md

  component: "masthead"
}) }}

## When to use this component

Use the masthead component on the landing page of non-transactional, product-orientated services.

Use this component to introduce users to your service and its primary benefits. Optionally provide a link to the primary call to action such as signing up or reading the instructions on how to get started.

## How it works

> The masthead uses your service’s brand colour (set using `$govuk-brand-colour`) and can incorporate the following:
> 
> - Phase banner
> - Breadcrumbs
> - Product name or proposition
> - Description of key benefits
> - Call to action button
> - A representative illustration

Reviewer: ```suggestion
The masthead uses your service’s brand colour (set using `$govuk-brand-colour`) and can include:

- a [phase banner](https://design-system.service.gov.uk/components/phase-banner/)
- [breadcrumbs](https://design-system.service.gov.uk/components/breadcrumbs/)
- the product name or proposition
- a description of key benefits
- a call to action button
- a representative illustration
```

Author: Lovely stuff.

> It should appear directly below the header, and take up the full-width of the page.

Reviewer: ```suggestion
It should appear directly below the header, and take up the full width of the page.
```

> To ensure there is no gap between the header and the masthead, create a modifying class to remove the bottom border from the header:

Reviewer: ```suggestion
To ensure there is no gap between the header and the masthead, create a modifier class to remove the bottom border from the header:
```

| :------ | :----- | :---------------------------------------------------------------------- |
| **src** | string | URL of image displayed on the right of the masthead on desktop layouts. |
| **alt** | array  | Alternative text for image.                                             |

## Research on this component

This component is based on the Masthead component used in the [Product Page Example](https://github.com/alphagov/product-page-example). This in turn was based on the product pages for:

- [GOV.UK Design System](https://design-system.service.gov.uk)
- [GOV.UK Forms](https://www.forms.service.gov.uk)
- [GOV.UK Notify](https://www.notifications.service.gov.uk)
- [GOV.UK One Login](https://www.sign-in.service.gov.uk)

> - [GOV.UK Pay](https://www.payments.service.gov.uk)

Reviewer: ```suggestion
- [GOV.UK Pay](https://www.payments.service.gov.uk)
- [GOV.UK GovWifi](https://www.wifi.service.gov.uk)
```

Versions of this component are currently being used on these services:

> - [Register trainee teachers](https://www.register-trainee-teachers.service.gov.uk)

Reviewer: Some more examples (too many?)

```suggestion
- [Register trainee teachers](https://www.register-trainee-teachers.service.gov.uk)
- [Gender pay gap service](https://gender-pay-gap.service.gov.uk)
- [Find a Learning Aim](https://submit-learner-data.service.gov.uk/find-a-learning-aim/)
- [Ethnicity facts and figures](https://www.ethnicity-facts-figures.service.gov.uk)
- [HMRC Developer Hub](https://developer.service.hmrc.gov.uk/api-documentation)
- [Find planning and housing data in England](https://www.planning.data.gov.uk/)
```

Author: It really ought to be a component in the Design System by now. It (or variations of it) are everywhere!
